presto-native-execution/presto_cpp/main/types/PrestoToVeloxConnector.h

```diff
@@ -0,0 +1,211 @@
+/*
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+#pragma once
+
+#include "PrestoToVeloxExpr.h"
+#include "presto_cpp/main/types/TypeParser.h"
+#include "presto_cpp/presto_protocol/ConnectorProtocol.h"
+#include "presto_cpp/presto_protocol/presto_protocol.h"
+#include "velox/connectors/Connector.h"
+#include "velox/connectors/hive/TableHandle.h"
+#include "velox/core/PlanNode.h"
+#include "velox/vector/ComplexVector.h"
+
+namespace facebook::presto {
+
+class PrestoToVeloxConnector;
+
+void registerPrestoToVeloxConnector(
+    std::unique_ptr<const PrestoToVeloxConnector> connector);
+
+const PrestoToVeloxConnector& getPrestoToVeloxConnector(
+    const std::string& connectorId);
+
+class PrestoToVeloxConnector {
+ public:
+  virtual ~PrestoToVeloxConnector() = default;
+
+  [[nodiscard]] const std::string& connectorId() const {
```

Comment:
let's not return a reference here.

Reply:
Could you please explain why a copy is preferred? What problem is there with leaving the decision to copy or not to copy to the caller?

Reply:
From an API standpoint, I see two problems. 
A user could change the private connectorId_ value by removing the const.
The lifetime of this variable might exceed that of the connector.
I don't think a copy here hurts.

Reply:
> A user could change the private connectorId_ value by removing the const.

You mean via a const_cast? Are we generally trying to make codebase resilient to `const_cast` mistakes? 

> The lifetime of this variable might exceed that of the connector.

Should the caller be responsible for creating a copy if the ownership change may occur? It seems to be quite common to return a reference / const reference to the managed objects (e.g.: map, vector, etc ..) and it's usually a responsibility of the caller side to think about ownership, lifecycle and whether a copy is necessary.

> I don't think a copy here hurts.

I don't think it hurts either. Copying a couple of bytes shouldn't be a problem. I'm quite new to the project and would love to gain better understand why certain design decisions are preferred over the other.

Reply:
> It seems to be quite common to return a reference / const reference to the managed objects

Fair enough. Let's stick to the existing patterns. I see we return a reference in the current API as well 
https://github.com/prestodb/presto/blob/2ff43e238a94eb182629ba04882d26ad21c3c8cc/presto-native-execution/presto_cpp/presto_protocol/Connectors.h#L50